app/models/effective/effective_datatable/resource.rb

```diff
@@ -169,12 +169,13 @@ def load_resource_search!
           end
 
           search = opts[:search]
+          option_text_method = search[:option_text_method] || :to_s
 
           # Parameterize collection
           if search[:collection].kind_of?(ActiveRecord::Relation)
-            search[:collection] = search[:collection].map { |obj| [obj.to_s, obj.to_param] }
+            search[:collection] = search[:collection].map { |obj| [obj.send(option_text_method), obj.to_param] }
           elsif search[:collection].kind_of?(Array) && search[:collection].first.kind_of?(ActiveRecord::Base)
-            search[:collection] = search[:collection].map { |obj| [obj.to_s, obj.to_param] }
+            search[:collection] = search[:collection].map { |obj| [obj.send(option_text_method), obj.to_param] }
```

Comment:
Could you change this to `public_send` rather than `send` for just a little bit more protection?

Shouldn't be calling private/protected methods here.